src/utils.js

```diff
@@ -520,7 +520,29 @@ export class DDGProxy {
     }
 }
 
+const maxCounter = new Map()
+function numberOfTimesDebugged (feature, message) {
+    let key
+    try {
+        // Could throw if message is not serializable
+        key = JSON.stringify({ feature, message })
```

Comment:
I'm guessing `JSON.stringify` in a loop could be taxing too. Besides, any difference in message would create a new key. Could we use a simpler key construction method? Perhaps just `feature`?

Reply:
Serializing isn't too concerning to me in comparison with swamping the messaging to another process a few times.

Feature seemed a little too coarse, the only concern I had is that the `jsException` is still serialized always (all others are gated with a `debug` flag).

- Maybe just up the limit to 5k and use feature is simpler though.

I actually removed this change but I think we should bring it back. If we're checking `debug` worst case we're just going to reintroduce a similar problem for people debugging with memory exhaustion.